PQ-2 Create Models and Liquibase changesets to create patient_queueing table

slubwama opened the pull request:
Create Models and Liquibase changesets to create patient_queueing table
https://issues.openmrs.org/browse/PQ-2
PR #2 has to be merged first before this PR is merged

slubwama committed:
Setup - Blank Module

inital Commit

WIP

slubwama committed:
Delete Non Base files

slubwama committed:
Create Models and Liquibase changesets to create patient_queueing table

slubwama committed:
 This is a combination of 2 commits.
 The first commit's message is:

 This is a combination of 2 commits.
 The first commit's message is:

 This is a combination of 2 commits.
 The first commit's message is:

Create Models and Liquibase changesets to create patient_queueing table

Setup - Blank Module

inital Commit

WIP

Delete Non Base files

 This is the 2nd commit message:

logging basing on recommendation

 This is the 2nd commit message:

Removing wild card import

 This is the 2nd commit message:

 This is a combination of 2 commits.
 The first commit's message is:

Merge branch 'master' of https://github.com/openmrs/openmrs-module-patientqueueing into PQ-2

# Conflicts:
#	api/src/main/java/org/openmrs/module/patientqueueing/PatientQueueingActivator.java
#	api/src/main/java/org/openmrs/module/patientqueueing/api/dao/PatientQueueingDao.java
#	api/src/main/resources/liquibase.xml
#	omod/src/test/java/org/openmrs/module/patientqueueing/AdminListExtensionTest.java
#	pom.xml

Setup - Blank Module

inital Commit

WIP

Delete Non Base files

logging basing on recommendation

Eliminating Wildcard import

Removing wild card import

Adding priority to Queueu

 This is the 2nd commit message:

Setup - Blank Module (#2)

inital Commit

WIP

Delete Non Base files

dkayiwa commented:
Did you see my comment on the ticket?

dkayiwa commented on api/src/main/java/org/openmrs/module/patientqueueing/api/dao/PatientQueueingDao.java:
```diff
@@ -0,0 +1,31 @@
+@Repository("patientqueueing.PatientQueueingDao")
+public class PatientQueueingDao {
+	
+	protected final Log log = LogFactory.getLog(getClass());
```
This is our recommended logging: https://wiki.openmrs.org/display/docs/Java+Conventions#JavaConventions-loggingLogging

dkayiwa commented on api/src/main/java/org/openmrs/module/patientqueueing/api/dao/PatientQueueingDao.java:
```diff
@@ -0,0 +1,31 @@
+import org.springframework.stereotype.Repository;
+
+@Repository("patientqueueing.PatientQueueingDao")
+public class PatientQueueingDao {
```
What is this class used for?

slubwama replied:
This will be used to handle saving and updating of the objects

dkayiwa commented on api/src/main/java/org/openmrs/module/patientqueueing/api/impl/PatientQueueingServiceImpl.java:
```diff
@@ -0,0 +1,35 @@
+
+public class PatientQueueingServiceImpl extends BaseOpenmrsService implements PatientQueueingService {
+	
+	PatientQueueingDao dao;
```
Shouldn't this be an interface?

slubwama replied:
There is an interface for this and this is the implementation

dkayiwa commented on api/src/main/java/org/openmrs/module/patientqueueing/model/PatientQueue.java:
```diff
@@ -0,0 +1,108 @@
+package org.openmrs.module.patientqueueing.model;
+
+import org.openmrs.*;
```
https://wiki.openmrs.org/display/docs/Java+Conventions#JavaConventions-Donotusewildcardimports

dkayiwa commented on api/src/main/java/org/openmrs/module/patientqueueing/model/PatientQueue.java:
```diff
@@ -0,0 +1,108 @@
+	}
+	
+	@Override
+	public User getCreator() {
```
Isn't this method in the base class?

dkayiwa commented on api/src/main/resources/liquibase.xml:
```diff
@@ -0,0 +1,94 @@
+            <column name="date_created" type="datetime"/>
+            <column name="changed_by" type="INT"/>
+            <column name="date_changed" type="datetime"/>
+            <column name="voided" type="TINYINT(3)"/>
```
What does it mean to void a patient queue?

slubwama replied:
It means that It can not be accessed by user without an exception to include voided. usecase when a queue is delete by a user It should just be disabled other than purge.

dkayiwa commented on omod/src/test/java/org/openmrs/module/patientqueueing/AdminListExtensionTest.java:
```diff
@@ -0,0 +1,48 @@
+import org.junit.Test;
+import org.openmrs.module.Extension;
+import org.openmrs.module.patientqueueing.extension.html.AdminList;
+import static org.hamcrest.Matchers.*;
```
https://wiki.openmrs.org/display/docs/Java+Conventions#JavaConventions-Donotusewildcardimports

dkayiwa commented on pom.xml:
```diff
@@ -0,0 +1,117 @@
+	<name>Patient Queueing</name>
+	<description>This Module Provides functionality to Queue Patient In a Hospital with a health care provider Dashboard to allow them to see the people in queue, message other providers and receive messages from other providers regarding patient care</description>
+
+	<url>https://wiki.openmrs.org/x/FQ4z</url>
```
What does the above link point to?

dkayiwa commented on pom.xml:
```diff
@@ -0,0 +1,117 @@
+			</dependency>
+			<!-- Importing versions from refapp distro -->
+            <dependency>
+                <groupId>org.openmrs.distro</groupId>
```
What is this dependency for?

slubwama replied:
@dkayiwa This is used to get dependencies from reference application which will be used in  module

slubwama committed:
logging basing on recommendation

slubwama committed:
Eliminating Wildcard import

slubwama committed:
Removing wild card import

slubwama committed:
Removing method that is a base method in OpenMRS

slubwama commented:
@dkayiwa requesting for more reviews

dkayiwa commented:
Can you look into these merge conflicts?

slubwama committed:
Adding priority to Queueu

slubwama committed:
Merge branch 'master' of https://github.com/openmrs/openmrs-module-patientqueueing into PQ-2

# Conflicts:
#	api/src/main/java/org/openmrs/module/patientqueueing/PatientQueueingActivator.java
#	api/src/main/java/org/openmrs/module/patientqueueing/api/dao/PatientQueueingDao.java
#	api/src/main/resources/liquibase.xml
#	omod/src/test/java/org/openmrs/module/patientqueueing/AdminListExtensionTest.java
#	pom.xml

Setup - Blank Module

inital Commit

WIP

Delete Non Base files

logging basing on recommendation

Eliminating Wildcard import

Removing wild card import

Adding priority to Queueu

slubwama committed:
 This is a combination of 2 commits.
 The first commit's message is:

Merge branch 'master' of https://github.com/openmrs/openmrs-module-patientqueueing into PQ-2

# Conflicts:
#	api/src/main/java/org/openmrs/module/patientqueueing/PatientQueueingActivator.java
#	api/src/main/java/org/openmrs/module/patientqueueing/api/dao/PatientQueueingDao.java
#	api/src/main/resources/liquibase.xml
#	omod/src/test/java/org/openmrs/module/patientqueueing/AdminListExtensionTest.java
#	pom.xml

Setup - Blank Module

inital Commit

WIP

Delete Non Base files

logging basing on recommendation

Eliminating Wildcard import

Removing wild card import

Adding priority to Queueu

 This is the 2nd commit message:

Setup - Blank Module (#2)

inital Commit

WIP

Delete Non Base files

slubwama committed:
Adding Queue Number that will be generated basing on a given criteria by implementations

slubwama commented:
Closed with Replacement #4
